src/app/config.rs

```diff
@@ -45,7 +45,10 @@ impl Config {
         }
     }
 
-    fn detect_patch_hub_config_file() -> Option<Config> {
+    /// Loads the configuration for patch-hub from the config file.
+    ///
+    /// Returns `None` if the config file is not found or if it's not a valid JSON.
+    fn load() -> Option<Config> {
```

Comment:
I think we should go for a middle-term here. IMHO, `load()` gets a little ambiguous, as one can understand "loading" in many ways; even more, if we consider that it is called as `Self::load()`. I think just adding the notion of loading from a file solves this. I favor something like `load_config_file`, but I can settle for something shorter like `load_file`.

Reply:
The documentation for this function is pretty explanatory. `Config::load_from_file` is the most explanatory name, but since we don't have any other config loading means, of course we are loading the config from a file. this way `Config::load` is short and explains what the function does. Further explainations are in the docs

Reply:
> The documentation for this function is pretty explanatory.

True, but the point is that even though the dev will find the explanation, it will demand hopping to the doc, and even if it is just a hover (we shouldn't asume he/she will be seeing it in a IDE, also), I prefer avoiding this extra work when we can.

> `Config::load_from_file` is the most explanatory name, but since we don't have any other config loading means, of course we are loading the config from a file. this way `Config::load` is short and explains what the function does.

IMHO, we are not easing the reader's work. Loading can mean a lot of things. Currently, we are only calling it in this context where it is clear that is from a file (this can mutate in the future, though), but we may expose this func and call it from `edit_config`, for example. The thing is, we save a few chars from `Config/Self::load_file()` to `Config/Self::load()`, whilst making it fairly more confusing.

Reply:
I'm really unable to see how `Config::load` may be confusing

Reply:
Remember that you already read the old name and the func implementation, so it is natural the meaning. When I was solving the merge conflicts, I accepted the incoming of this part by mistake, and genuinely was confused about what the load meant. Had to scroll back up to realize that it was the old detect func. `load` isn't bad, but it won't hurt 5 chars, since we already are cutting many chars comparing to the old name.